src/hotspot/share/opto/gcm.cpp

```diff
@@ -662,23 +663,65 @@ class DefUseMemStatesQueue : public StackObj {
   }
 };
 
-//--------------------------insert_anti_dependences---------------------------
-// A load may need to witness memory that nearby stores can overwrite.
-// For each nearby store, either insert an "anti-dependence" edge
-// from the load to the store, or else move LCA upward to force the
-// load to (eventually) be scheduled in a block above the store.
+//------------------------raise_above_anti_dependences---------------------------
+// The argument load has a current scheduling range in the dominator tree that
```

Comment:
COMMENT: Could you start this long comment with a one-sentence summary of what the function does? I.e. something like "Enforce a scheduling of the given load where its input memory state is not overwritten by an anti-dependent store".

Reply:
Sure! I went with
```
// Enforce a scheduling of the argument load that ensures anti-dependent stores
// do not overwrite the load's input memory state before the load executes.
```